client/battle/CBattleInterface.cpp

```diff
@@ -3275,19 +3275,31 @@ void CBattleInterface::showAliveStacks(SDL_Surface *to, std::vector<const CStack
 {
 	auto isAmountBoxVisible = [&](const CStack *stack) -> bool
 	{
-		if (stack->hasBonusOfType(Bonus::SIEGE_WEAPON)) // siege weapons are always singular
+		if(stack->hasBonusOfType(Bonus::SIEGE_WEAPON)) // siege weapons are always singular

```

Comment:
Well, I guess we can change it in future if needed. For example with future scripting support we may try (or let others try) to reimplement some original WoG war machine scripts that handle stacked war machines support.

Or maybe you have any idea what to do with this code now? We can add "&& stack->count ==1" to condition to hide box only when there is 1 war machine in stack.